docs: update trg 1-2 add install example and best practices (#559)
eclipse-tractusx/eclipse-tractusx.github.io: docs/release/trg-1/trg-1-2.md

| Update | 13-Dez-2023  | add best practices and an example, rephrase the why |
| Active | 07-Mar-2023  |                                                     |
| Draft  | 13-Sept-2022 |                                                     |

## Why

If a repository contains code which needs to be configured/setup or installed, the `README.md` file should cover this information.

If these instructions are too long, they can be moved into a dedicated `INSTALL.md` file.

Having a proper `INSTALL.md` file is crucial for ensuring a seamless and user-friendly experience for the product-users.

In conclusion, a proper `INSTALL.md` file should be mainly adapted to end-users, but also developers. It is not a technical necessity.
It's a strategic investment in user satisfaction, community engagement, and the overall success of the product. It reflects a commitment to inclusivity and usability, ensuring that a diverse user base can effortlessly install and enjoy the benefits of your product.

## Description

Create a `INSTALL.md` file in the main folder of the repository and add all installation relevant information.

## Best Practices

> a well structured installation guide includes

Reviewer: Start sentence with capital letter and fix markdown error as list did not have blank line before itself.

```suggestion
A well structured installation guide includes

```

Author: https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/pull/559/commits/9d85c5b5db5e8001ac155a3373d25b195300ffcf

title: TRG 1.02 - INSTALL.md
---

| Status | Created      | Post-History                                        |
|--------|--------------|-----------------------------------------------------|
| Update | 13-Dez-2023  | add best practices and an example, rephrase the why |
| Active | 07-Mar-2023  |                                                     |
| Draft  | 13-Sept-2022 |                                                     |

## Why

If a repository contains code which needs to be configured/setup or installed, the `README.md` file should cover this information.

If these instructions are too long, they can be moved into a dedicated `INSTALL.md` file.

> Having a proper "install.md" file is crucial for ensuring a seamless and user-friendly experience for the product-users.

Reviewer 2: `INSTALL.md` instead of "install.md" -> keep the same file name and formatting

Author: [eb1f28d](https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/pull/559/commits/eb1f28dc0ff569251dc8c340b33e7ffa4db250d6)

> Having a proper "install.md" file is crucial for ensuring a seamless and user-friendly experience for the product-users.

Author: for the moment regarding to a dedicated `INSTALL.md` yes, but you're right this should be a general info for all documentation for the products
[eb1f28d](https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/pull/559/commits/eb1f28dc0ff569251dc8c340b33e7ffa4db250d6)

> Having a proper "install.md" file is crucial for ensuring a seamless and user-friendly experience for the product-users.

Author: like Having a proper documentation is crucial for ensuring a seamless and user-friendly experience for the product-users. 😉

> In conclusion, a proper install.md file tailored for end-users, but also developers. It is not a technical necessity, it's a strategic investment in user satisfaction, community engagement, and the overall success of the product. It reflects a commitment to inclusivity and usability, ensuring that a diverse user base can effortlessly install and enjoy the benefits of your product.

Reviewer 2: `INSTALL.md` instead of install.md

Author: [eb1f28d](https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/pull/559/commits/eb1f28dc0ff569251dc8c340b33e7ffa4db250d6)

> In conclusion, a proper install.md file tailored for end-users, but also developers. It is not a technical necessity, it's a strategic investment in user satisfaction, community engagement, and the overall success of the product. It reflects a commitment to inclusivity and usability, ensuring that a diverse user base can effortlessly install and enjoy the benefits of your product.

Reviewer 2: is tailored?

Author: [eb1f28d](https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/pull/559/commits/eb1f28dc0ff569251dc8c340b33e7ffa4db250d6)

> In conclusion, a proper install.md file tailored for end-users, but also developers. It is not a technical necessity, it's a strategic investment in user satisfaction, community engagement, and the overall success of the product. It reflects a commitment to inclusivity and usability, ensuring that a diverse user base can effortlessly install and enjoy the benefits of your product.

Reviewer 2: It is not a technical necessity -> dot instead of comma I guess. Seems like a new sentence starting after it

Author: [eb1f28d](https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/pull/559/commits/eb1f28dc0ff569251dc8c340b33e7ffa4db250d6)

## Description

Create a `INSTALL.md` file in the main folder of the repository and add all installation relevant information.

## Best Practices

- add a clear installing guide through our official helm chart repo
- add table of content of your installation guide
- add a numeric step-by-step installation guide with command copy options
- add prerequisites section for the installation
- add a reference your helm chart README.md file
- add guidance on how to verify the installation
- add local deployment advisories
- add a troubleshooting section
- add a FAQ section
- add additional resources information
- add uninstallation guide
- add upgrade guide
- add security considerations
- add contact information for support or feedback

> - add a thankful note for the user for using the product

Reviewer 2: Hmm not sure about that. seems a bit much 😁

Author: [eb1f28d](https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/pull/559/commits/eb1f28dc0ff569251dc8c340b33e7ffa4db250d6)

> - add a thankful note for the user for using the product

Reviewer 3: Just a suggestion, it may look cleaner removing 'add' from each line, should be enough to state somewhere above to include following sections.

Author: [eb1f28d](https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/pull/559/commits/eb1f28dc0ff569251dc8c340b33e7ffa4db250d6)

> ## Example Installation Guide

Reviewer 3: Provided example doesn't cover all items from 'Best Practices' list, maybe we don't really need that many? :)

Author: Should ne a little example fir install instructions and the rest are advises

Author: [eb1f28d](https://github.com/eclipse-tractusx/eclipse-tractusx.github.io/pull/559/commits/eb1f28dc0ff569251dc8c340b33e7ffa4db250d6) 

add examplary helm install section to make it clear that this is just a partiacl example

If these instructions are too long, they can be moved into a dedicated `INSTALL.md` file.

Having a proper `INSTALL.md` file is crucial for ensuring a seamless and user-friendly experience for the product-users.

In conclusion, a proper `INSTALL.md` file should be mainly adapted to end-users, but also developers. It is not a technical necessity.
It's a strategic investment in user satisfaction, community engagement, and the overall success of the product. It reflects a commitment to inclusivity and usability, ensuring that a diverse user base can effortlessly install and enjoy the benefits of your product.

## Description

Create a `INSTALL.md` file in the main folder of the repository and add all installation relevant information.

## Best Practices

> - a clear installing guide through our official helm chart repo
> - table of content of your installation guide
> - prerequisites section for the installation
> - a numeric step-by-step installation guide with command copy options
> - a reference your helm chart `README.md` file
> - install verification / uninstallation / upgrade guide if needed local deployment advisories
> - a FAQ and troubleshooting section
> - additional resources information

Reviewer 4: How about something like:
```suggestion
a well structured installation guide includes
- the prerequisites
- a step-by-step installation instruction which can be copy+pasted directly
  - using only the charts of our official [helm chart repository](https://github.com/eclipse-tractusx/charts)
- a link to your helm chart `README.md` file
- guides on verification, upgrade and uninstallation (and if needed: local deployment/testing specifics)
- a FAQ and troubleshooting section
- links to additional resources / information
```

> - contact information for support or feedback

Reviewer 4: Is this even necessary? We use GH isses anyways by default
```suggestion
```

Author: Jep we so, but maybe other Teams follow an other strategy and could link more information for support
